storage-proofs-update/src/circuit.rs

```diff
@@ -97,69 +96,48 @@ impl PublicInputs {
     }
 }
 
-pub struct ChallengeProof<TreeR>
+#[derive(Clone)]
+pub struct ChallengeProof<U, V, W>
 where
-    TreeR: MerkleTreeTrait<Hasher = TreeRHasher>,
+    U: PoseidonArity,
+    V: PoseidonArity,
+    W: PoseidonArity,
 {
     pub leaf_r_old: Option<Fr>,
     pub path_r_old: Vec<Vec<Option<Fr>>>,
     pub leaf_d_new: Option<Fr>,
     pub path_d_new: Vec<Vec<Option<Fr>>>,
     pub leaf_r_new: Option<Fr>,
     pub path_r_new: Vec<Vec<Option<Fr>>>,
-    pub _tree_r: PhantomData<TreeR>,
-}
-
-// Implement `Clone` by hand because `MerkleTreeTrait` does not implement `Clone`.
```

Comment:
Note: we needed to hand implement `Clone` because `TreeR: MerkleTreeTrait` didn't; removing the `TreeR` trait bound means that we can derive `Clone`